helpers/helpers.go

```diff
@@ -321,22 +322,30 @@ func RunCommandOutputEnv(cmdname string, args []string, envs []string) (*bytes.B
 	cmd.Stdout = &outBuf
 	cmd.Stderr = &errBuf
 	cmd.Env = append(os.Environ(), envs...)
-	err := cmd.Run()
+	runError := cmd.Run()
 
-	if err != nil {
+	if runError != nil {
 		var buf bytes.Buffer
-		fmt.Fprintf(&buf, "failed to execute %s", strings.Join(cmd.Args, " "))
+		if _, err := fmt.Fprintf(&buf, "failed to execute %s", strings.Join(cmd.Args, " ")); err != nil {
```

Comment:
This will also just print several of the same warning to stderr as it goes down this function.
Since we return an error at the end of this function, I also don't care if writing to the buffer fails. If writing to a bytes.Buffer fails, probably the whole system is hosed anyway to actually continue mixing. I think writing to actual *Files* is the only place we really need to check for errors on writing, because those can fail for a variety of subtle reasons.

Reply:
You could change all the `fmt.Fprintf()` calls to `log.Printf()` setting the output to be buf, or making a logger var using buf as the io.Writer, and then we get to remove all the "Warning" lines and change all the Fprintf() lines to 1 line since log.Prints don't return an error to check.

Reply:
> If writing to a bytes.Buffer fails, probably the whole system is hosed anyway to actually continue mixing.

That's a mighty big assumption to make. And if you are wrong...

> I think writing to actual Files is the only place we really need to check for errors on writing

writing to the file could actually pass but it may be writing a corrupted buffer.

Reply:
I personally like printing the errors as warnings here.

Reply:
> That's a mighty big assumption to make. And if you are wrong...
We make it all the time in swupd! <function>_or_die()

> writing to the file could actually pass but it may be writing a corrupted buffer.

What does it mean to corrupt your buffer in Go?

Reply:
> We make it all the time in swupd! function_or_die()

No we make the opposite assumption than you are. If a write to memory fails we just die. You are saying allow it to happen and just write to the file anyways.

Maybe we should go the swupd route and just fail immediately then? Is that what you are suggesting? I'd be ok with that.

> What does it mean to corrupt your buffer in Go?

idk, since I'm not sure why a buffer write would fail other than oom. Seems like a good argument for just failing at the buffer write to me.

Reply:
Ok so digging through the go source code looks like an `Fprintf` on a `bytes.Buffer` will **never** return an `error`.

The `Fprintf` just calls the `Write` method of the passed `io.Writer`, in this case a `bytes.Buffer`:
https://golang.org/src/fmt/print.go?s=4979:5056#L176
```go
func Fprintf(w io.Writer, format string, a ...interface{}) (n int, err error) {
	p := newPrinter()
	p.doPrintf(format, a)
	n, err = w.Write(p.buf)
	p.free()
	return
}
```
You can see it returns the `err` from `w.Write`, but for a `bytes.Buffer.Write` `err` will always be `nil`:
https://golang.org/src/bytes/buffer.go?s=5775:5826#L160
```go
func (b *Buffer) Write(p []byte) (n int, err error) {
	b.lastRead = opInvalid
	m, ok := b.tryGrowByReslice(len(p))
	if !ok {
		m = b.grow(len(p))
	}
	return copy(b.buf[m:], p), nil
}
```

So we definitely can throw away the error when writing to a buffer.